_posts/2018-12-20-data-modelling-many-to-many.adoc

```diff
@@ -0,0 +1,179 @@
+---
+compatible_cuba_versions: 7.0
+project_id: cuba-petclinic-data-model-many-to-many
+permalink: data-modelling-many-to-many-association
+---
+= Data modelling: Many-to-Many association
+:showtitle:
+:sectlinks:
+:sectanchors:
+:page-navtitle: Data modelling: Many-to-Many association
+:page-excerpt: Data modelling guide: what are Many-to-Many associations and when to use them
+:page-root: ../../../
+:project_id: cuba-petclinic-data-model-many-to-many
+:java_version: 1.8
+:cuba_version: 7.0
+
+The many-to-many association implies that multiple records in one entity are related to multiple records in another entity. The joining table will store primary keys of both related entities. Optionally, this table may contain additional columns.
+
+Depending on whether additional fields in the joining table are needed, it can be implemented via a many to many association with an additional entity. The following examples illustrate both approaches.
+
+== What will be build
+
+This guide enhances the https://github.com/cuba-platform/cuba-petclinic[CUBA petclinic] example to show different use cases of many to many associations. In particular, the following use cases will be covered:
+
+* `Vet` <--> `Specialty` is modelled as a direct Many-to-Many association and the UI is updated accordingly
+* `Pet` <--> `InsuranceCompany` is modelled as an indirect Many-to-Many association with an explicit entity to store the validity range
+
+=== Final Application
```

Comment:
ok, will do. In case of `What will be built` it would mean: `What Will Be Built` according to https://capitalizemytitle.com/. This looks a little bit wired (but perhaps I'm just not used to it). How do you handle it in other stuff like the docs?

Reply:
AFAIK the "Title Case" is a standard for English and we use it throughout the documentation (while in the Russian version of the docs we use "First word capitalized" as there is no such rule for Russian text). 
So perhaps it would be better to use the title case in all guides too.

Reply:
done